src/ILCompiler.Compiler/src/Compiler/Compilation.cs

```diff
@@ -204,9 +205,24 @@ private void AddCompilationRoots()
                 AddCompilationRootsForRuntimeExports(module);
            }
 
+            AddCompilationRootsForBootstrap();
```

Comment:
I would tend to agree. The thing we are replacing is a bootstrapper, but the new thing is simply the startup code.